packages/db/src/models/granule.ts

```diff
@@ -5,30 +5,79 @@ import { tableNames } from '../tables';
 
 import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 
+import { GranuleExecutionHistoryPgModel } from './granule-execution-history';
+
 export default class GranulePgModel extends BasePgModel<PostgresGranule, PostgresGranuleRecord> {
   constructor() {
     super({
       tableName: tableNames.granules,
     });
   }
 
-  upsert(
+  async createWithExecutionHistory(
+    knexOrTrx: Knex | Knex.Transaction,
+    item: PostgresGranule,
+    executionCumulusId: number,
+    granuleExecutionHistoryPgModel = new GranuleExecutionHistoryPgModel()
+  ) {
+    const [granuleCumulusId] = await this.create(knexOrTrx, item);
+    await granuleExecutionHistoryPgModel.create(
```

Comment:
Do you mean when we do a granule delete in general to make sure we handle deleting the history? If so, I haven't even implemented any of that and it's partially blocked by #2057 but it's probably also worth doing as part of this PR

Reply:
TODO: make sure to handle deleting granule/execution links when deleting granules in API/wherever

Reply:
⚠️ This is a fairly substantial change: bc9a6af13b3947ff3a1fb9e2a389614f4ec4cd6d

I updated the FK columns for the `granules_executions` table to use `ON CASCADE DELETE`, meaning that when you delete a granule record or execution record, then the rows referencing those records in granules_executions are automatically delete. This would seem to be exactly what we want, but it also seemed vastly preferable to the other options:

- Lookup and manually delete records from the join table, which requires knowing a granule/execution PK at deletion time, which shouldn't be required and would enforce the use of `.get` calls in our deletion logic